File: renders/habitats/WaterHabitat.cpp
/* NIM/Nama  : 13515AAA / AAA       */
/* File      : Habitat.cpp              */

#include "WaterHabitat.h"

using namespace std;

WaterHabitat::WaterHabitat() : id('#'), color(WHITE)
{
}
WaterHabitat::WaterHabitat(const WaterHabitat&) : id('#'), color(WHITE)
{
}
WaterHabitat::~WaterHabitat()
{
}
WaterHabitat& WaterHabitat::operator=(const WaterHabitat&)
{
}
WaterHabitat* WaterHabitat::clone() const
{
	return new WaterHabitat(*this);
}

File: src/habitat/water/WaterHabitat.java
/* NIM/Nama : 13515078 / Veren Iliana Kurniadi */
/* File     : WaterHabitat.java                */

package habitat.water;

import habitat.Habitat;

public class WaterHabitat extends Habitat {
  /**
   * Constructor.
   */
  public WaterHabitat() {
    super();
  }
  
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar.
   * @return id bertipe char
   */
  public char render() {
    return 'W';
  }
  
  /**
   * Mengembalikan warna untuk ditampilkan ke layar.
   * @return color warna renderable
   */
  public String renderWithColor() {
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
